### PR TITLE
LPS-177794 [Raylife AP]  - Create data consumer

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[LPS 177794 ](https://issues.liferay.com/browse/LPS-174067)

### CODE REVIEW

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Review by joanagrego — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+		Application: 'Application ',
+		Claim: 'Claim',
+		Policy: 'Policy',
+	};
+
+	const creatRoute = (externalReferenceCode: string) => {
+		const link = `ap/app-details?externalReferenceCode=${externalReferenceCode}`;
+
+		return link;
+	};
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/notification.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ const headlessAPI = 'o/headless-user-notification/v1.0';
 
 export function getUserNotification(pageSize: number, page: number) {
 	return axios.get(
-		`${headlessAPI}/my-user-notifications/?&pageSize=${pageSize}&page=${page}`
+		`${headlessAPI}/my-user-notifications/?&pageSize=${pageSize}&page=${page}&sort=dateCreated:desc`
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/notification.ts:21]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/Application.ts [hunk 3]
```diff
@@ -129,6 +129,12 @@ export function getApplicationsById(id: number) {
 	return axios.get(`${DeliveryAPI}/?filter=id eq '${id}'`);
 }
 
+export function getExternalReferenceCodeById(id: number) {
+	return axios.get(
+		`${DeliveryAPI}/?filter=id eq '${id}'&fields=externalReferenceCode`
+	);
+}
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/Application.ts:136]: Good suggestion!
I  add optional  a parameter in `getApplicationsById()`
Thx

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+import {PostType} from './postTypes';
+
+const initialPagination = {page: 1, pageSize: 2, totalCount: 0};
+
+const NotificationSidebar: React.FC = () => {
+	const [posts, setPosts] = useState<any[]>([]);
+	const [totalCount, setTotalCount] = useState<number>(
+		initialPagination.totalCount
+	);
+	const [arrayOfLinks, setArrayOfLinks] = useState<any[]>([]);
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:35]: Done

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+	return (
+		<div className="notification-container">
+			{!arrayOfLinks.length && (
+				<p className="align-items-center d-flex justify-content-center pt-8 vh-100">
+					No notifications
+				</p>
+			)}
+
+			{!!arrayOfLinks.length && (
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:162]: Nice!!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+	const notificationCategory = {
+		Application: 'Application ',
+	};
+
+	const creatRoute = (externalReferenceCode: string, entity: string) => {
+		const currentSiteName = getLiferaySiteName();
+		const link = `${currentSiteName}/${entity}?externalReferenceCode=${externalReferenceCode}`;
+
+		return link;
+	};
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:49]: Could you add this to a component file? 
```suggestion
	const createUrl = (externalReferenceCode: string, entity: string) => {
		const currentSiteName = getLiferaySiteName();
		const link = `${currentSiteName}/${entity}?externalReferenceCode=${externalReferenceCode}`;

		return link;
	};
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+import {PostType} from './postTypes';
+
+const initialPagination = {order: 'desc', page: 1, pageSize: 2, totalCount: 0};
+
+const NotificationSidebar: React.FC = () => {
+	const [posts, setPosts] = useState<PostType[]>([]);
+	const [totalCount, setTotalCount] = useState<number>(
+		initialPagination.totalCount
+	);
+	const [postsWithLinks, setArrayOfLinks] = useState<PostType[]>([]);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:36]:
```suggestion
	const [postsWithLinks, setPostsWithLinks] = useState<PostType[]>([]);
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+	const [page, setPage] = useState<number>(initialPagination.page);
+	const hasMorePostsToLoad = posts.length < totalCount;
+
+	const notificationCategory = {
+		Application: 'Application ',
+	};
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:42]: Please change it to a const.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+			)}
+
+			{!!postsWithLinks && (
+				<div className="vh-100">
+					{postsWithLinks.map((item: PostType, _index: number) => (
+						<div
+							className={classNames({
+								'post-container-unread align-items-center d-flex justify-content-center position-relative bubble-unread': !item.read,
+							})}
+							key={item.id}
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:170]:
```suggestion
							key={index}
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+	const generateLinks = async () => {
+		const newLinks = await Promise.all(
+			posts.map(async (post) => {
+				const postId = extractNumber(post.message);
+				const isMatchingApplication = post.message?.includes(
+					notificationCategory.Application + postId
+				);
+
+				const genericRoute = '#!';
+				if (isMatchingApplication) {
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:119]: Please add an empty line here.
```suggestion

				if (isMatchingApplication) {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+				`Error fetching external reference code for ID ${id}: ${error}`
+			);
+			throw error;
+		}
+	}
+
+	const generateLinks = async () => {
+		const newLinks = await Promise.all(
+			posts.map(async (post) => {
+				const postId = extractNumber(post.message);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:113]:
```suggestion
				const postId = extractNumber(post.message as string);
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/Application.ts [hunk 12]
```diff
@@ -125,7 +125,13 @@ export function getApplicationByExternalReferenceCode(
...
 
-export function getApplicationsById(id: number) {
+export function getApplicationsById(id: number, fields?: string) {
+	if (fields) {
+		return axios.get(
+			`${DeliveryAPI}/?filter=id eq '${id}'&fields=${fields}`
+		);
+	}
+
 	return axios.get(`${DeliveryAPI}/?filter=id eq '${id}'`);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/Application.ts:135]: Very good!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/notification.ts [hunk 13]
```diff
@@ -16,9 +16,13 @@ import {axios} from './liferay/api';
...
+export function getUserNotification(
+	pageSize: number,
+	page: number,
+	order: string
+) {
 	return axios.get(
-		`${headlessAPI}/my-user-notifications/?&pageSize=${pageSize}&page=${page}`
+		`${headlessAPI}/my-user-notifications/?&pageSize=${pageSize}&page=${page}&sort=dateCreated:${order}`
 	);
 }
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/notification.ts:27]: The "boy-scout rule': 'Always leave the code better than you found it. '

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-fragments/fragments/layout/index.css [hunk 14]
```diff
@@ -197,12 +197,35 @@
...
 	transition: all 0.15s ease-in;
 	width: 0;
 }
+
+.layout-container .notification-sidebar {
+	height: 100%;
+	opacity: 0;
+	overflow-x: none;
+	overflow-y: auto;
+	width: 0px;
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-fragments/fragments/layout/index.css:212]: Is it possible to use Bootstrap here?

**Review by joanagrego — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx (hunk 6 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx:49]: Done, thx!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/notification.ts (hunk 13 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/services/notification.ts:27]: Done!

**Comment by joanagrego** [comment]
ci:forward

**Pull request closed without merging**